code/game/objects/items/mail.dm

```diff
@@ -332,3 +334,166 @@
+	icon_state = "mail_counterfeit_device" // placeholder for now
+
+/obj/item/storage/mail_counterfeit_device/Initialize(mapload)
+	. = ..()
+	atom_storage.max_slots = 1
+	atom_storage.allow_big_nesting = TRUE
+	atom_storage.max_specific_storage = WEIGHT_CLASS_NORMAL
+
+/obj/item/storage/mail_counterfeit_device/attack_self(mob/user, modifiers)
+	var/mail_type = tgui_alert(user, "Is it gonna be an envelope or a normal mail?", "Mail Counterfeiting", list("Mail", "Envelope"))
+	if(isnull(mail_type))
+		return FALSE
+	if(!(src in user.contents))
+		return FALSE
+	mail_type = lowertext(mail_type)
+
+	var/mail_armed = tgui_alert(user, "Is it gonna be armed?", "Mail Counterfeiting", list("Yes", "No"))
+	if(isnull(mail_armed))
+		return FALSE
+	if(!(src in user.contents))
+		return FALSE
+
+	if(mail_armed == "Yes")
+		mail_armed = TRUE
+	else
+		mail_armed = FALSE
+
+	var/list/mail_recipients = list()
+	var/list/mail_recipients_input_list = list("0# Anyone")
+	var/iterator = 1
+	for(var/mob/living/carbon/human/human in GLOB.human_list)
+		// Skip everyone who is not part of the crew...
+		if(isnull(human.mind))
+			continue
+		if(!(human.mind.assigned_role.job_flags & JOB_CREW_MEMBER))
+			continue
+		mail_recipients += human.mind
+		mail_recipients_input_list += "[iterator]# [human.name]"
+		iterator++
+	
+	var/recipient = tgui_input_list(user, "Choose a recipient", "Mail Counterfeiting", mail_recipients_input_list)
+	if(isnull(recipient))
+		return FALSE
+	if(!(src in user.contents))
+		return FALSE
+	
+	
+	var/index = text2num(copytext(recipient, 1, findtext(recipient, "#")))
+	if(index == 0)
+		var/mail_name = tgui_input_text(user, "Enter mail title or leave it blank to get a default one.", "Mail Counterfeiting")
+		if(!reject_bad_text(mail_name, ascii_only = FALSE))
+			mail_name = mail_type
+		if(!(src in user.contents))
+			return FALSE
+		
+		
+		var/obj/item/mail/traitor/shady_mail
+		if(mail_type == "mail")
+			shady_mail = new /obj/item/mail/traitor()
+		else
+			shady_mail = new /obj/item/mail/traitor/envelope()
+		
+		atom_storage.hide_contents(user)
+		user.temporarilyRemoveItemFromInventory(src, TRUE)
+		shady_mail.contents += contents
+		shady_mail.name = mail_name
+		shady_mail.armed = mail_armed
+		shady_mail.madeby = user
+		user.put_in_hands(shady_mail)
+	else
+		if(!(src in user.contents))
+			return FALSE
+
+		var/obj/item/mail/traitor/shady_mail
+		if(mail_type == "mail")
+			shady_mail = new /obj/item/mail/traitor()
+		else
+			shady_mail = new /obj/item/mail/traitor/envelope()
+		
+		atom_storage.hide_contents(user)
+		user.temporarilyRemoveItemFromInventory(src, TRUE)
+		shady_mail.contents += contents
+		shady_mail.armed = mail_armed
+		shady_mail.madeby = user
+		shady_mail.initialize_for_recipient(mail_recipients[index])
+		user.put_in_hands(shady_mail)
+	qdel(src)
+
+// =======================
+// Admin stuff down here
+// for events or shitspawn
+// purposes.
+// =======================
+
```

Comment:
Normal doc comment, do not do ASCII art.

Reply:
Okay.